pipeline/splicing/calcVarPriors.py

```diff
@@ -733,65 +770,161 @@ def getMaxMaxEntScanScoreSlidingWindowSNS(variant, windowSize=9, exonicPortionSi
     '''
-    Given a variant, checks if variant is in an exon
+    Given a variant, determines scores for closest reference splice sequence
+    deNovoOffset refers to difference between de novo acceptor length and exonic portion size
+       If donor = True, looks for closest splice donor sequence
+       If donor = False, looks for closest splice acceptor sequence
+       If deNovo = True, accomodates for de novo splicing
+         *Note only use argument deNovo=True in this function if donor=False
+         *Function will not return correct sequence if donor=True and deNovo=True
     If exonic variant, returns a dictionary containing:
-       MaxEntScan score and z-score for reference subsequent donor sequence
+       MaxEntScan score and z-score for reference closest splice sequence
+    If variant located in referene splice site, returns a dictionary containing:
+       MaxEntScan score and z-score for that reference splice site sequence
+    deNovoDonorInRefAcc = False if NOT checking for de novo splice donor sites in reference splice acceptor sites
+    deNovoDonorInRefAcc = True if checking for de novo splice donor sites in reference splice acceptor sites 
     '''
     varGenPos = int(variant["Pos"])
-    inExon = varInExon(variant)
-    if inExon == True:
-        varChrom = getVarChrom(variant)
-        varStrand = getVarStrand(variant)
+    varChrom = getVarChrom(variant)
+    if varInExon(variant) == True and deNovo == False:
         exonNumber = getVarExonNumberSNS(variant)
-        refSpliceDonorBounds = getRefSpliceDonorBoundaries(variant)
-        subDonorBounds = refSpliceDonorBounds[exonNumber]
-        refSeq = getFastaSeq(varChrom, subDonorBounds["donorStart"], subDonorBounds["donorEnd"])
-        subMaxEntScanScore = runMaxEntScan(refSeq, donor=True)
-        subZScore = getZScore(subMaxEntScanScore, donor=True)
-        return {"maxEntScanScore": subMaxEntScanScore,
-                "zScore": subZScore}
+        exonName = exonNumber
+        if donor == True:
+            refSpliceDonorBounds = getRefSpliceDonorBoundaries(variant, STD_DONOR_INTRONIC_LENGTH, STD_DONOR_EXONIC_LENGTH)
+            closestSpliceBounds = refSpliceDonorBounds[exonNumber]
+        else:
+            refSpliceAccBounds = getSpliceAcceptorBoundaries(variant, STD_ACC_INTRONIC_LENGTH, STD_ACC_EXONIC_LENGTH)
+            closestSpliceBounds = refSpliceAccBounds[exonNumber]
+    if varInSpliceRegion(variant, donor=donor, deNovo=deNovo) == True and deNovoDonorInRefAcc == False:
+        closestSpliceBounds = getVarSpliceRegionBounds(variant, donor=donor, deNovo=deNovo)
+        exonName = closestSpliceBounds["exonName"]
+    if donor == True:
+        refSeq = getFastaSeq(varChrom, closestSpliceBounds["donorStart"], closestSpliceBounds["donorEnd"])
+    else:
+        # acceptorEnd +- deNovoOffset because deNovo splice acceptor region is deNovoOffset bp longer than reference splice acceptor region
```

Comment:
It is not independent of strand. For the plus strand it is acceptorEnd - deNovoOffset because the genomic position increases from left to right so this reduces the refSeq to it's correct length (23 bp). For the minus strand it is acceptorEnd + deNovoOffset because the genomic position decreases from right to left so this again reduces refSeq to 23 bp. I can include a comment explaining this if now or in the next PR if that would be helpful

Reply:
A comment in the next PR will be fine.  I'm glad to hear that this was a false alarm!